File: software/examples/c/example_simple.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_line.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	Line l;
	line_create(&l, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Get current reflectivity
	uint16_t reflectivity;
	if(line_get_reflectivity(&l, &reflectivity) < 0) {
		fprintf(stderr, "Could not get reflectivity, probably timeout\n");
		return 1;
	}

	printf("Reflectivity: %d\n", reflectivity);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
	"github.com/tinkerforge/go-api-bindings/line_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Line Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	l, _ := line_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current reflectivity.
	reflectivity, _ := l.GetReflectivity()
	fmt.Printf("Reflectivity: \n", reflectivity)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
